core_lib_generator/core_lib_template_generator.py

```diff
@@ -0,0 +1,235 @@
+import enum
+
+from core_lib.helpers.shell_utils import input_enum, input_str, input_yes_no, input_int, input_list
+
+
+class DBTypes(enum.Enum):
+    __order__ = 'SQLite Postgresql MySQL Oracle MSSQL Firebird Sybase'
+    SQLite = 1
+    Postgresql = 2
+    MySQL = 3
+    Oracle = 4
+    MSSQL = 5
+    Firebird = 6
+    Sybase = 7
+
+
+class DBDatatypes(enum.Enum):
+    __order__ = 'INTEGER VARCHAR DATETIME DATE BOOLEAN'
+    INTEGER = 1
+    VARCHAR = 2
+    DATETIME = 3
+    DATE = 4
+    BOOLEAN = 5
+
+
+class CacheTypes(enum.Enum):
+    __order__ = 'Memcached Memory Empty'
+    Memcached = 1
+    Memory = 2
+    Empty = 3
+
+
+class DataAccessTypes(enum.Enum):
+    __order__ = 'CRUD SoftDelete SoftDeleteToken CRUDSoftDelete CRUDSoftDeleteToken'
+    CRUD = 1
+    SoftDelete = 2
+    SoftDeleteToken = 3
+    CRUDSoftDelete = 4
+    CRUDSoftDeleteToken = 5
+
+
+def generate_db_template() -> dict:
+    db_template = {}
+    add_db = True
+    while add_db:
+        db_name = input_str('What is the name of the DB connection?')
+        while db_name in db_template:
+            db_name = input_str('What is the name of the DB connection?')
+
+        db_type = input_enum(
+            DBTypes, 'for DB type'
+        )
+
+        if db_type == DBTypes.SQLite.value:
+            in_memory = input_yes_no('Do you want the SQLite DB in memory?', True)
+            if in_memory:
+                db_log_queries = input_yes_no('Do you want to log queries?', False)
+                db_create = input_yes_no('Do you want create Database?', True)
+                db_pool_recycle = input_int('Enter the pool recycle time: ', 3200)
+                db_pool_pre_ping = input_yes_no('Do you want to set pool pre ping?', False)
+                print('\nSQLite created in memory.')
+                add_db = input_yes_no('\nDo you want to add another DB connection?', False)
+                db_template[db_name.lower()] = {
+                    'log_queries': db_log_queries,
+                    'create_db': db_create,
+                    'session': {
+                        'pool_recycle': db_pool_recycle,
+                        'pool_pre_ping': db_pool_pre_ping,
+                    },
+                    'url': {
+                        'protocol': DBTypes(db_type).name.lower(),
+                    },
+                }
+                continue
+        default_db_ports = {

```

Comment:
define this outside the function under the DBTypes enum. no need to recreated every function call